IanGRap/Squish
# Lobby Changes Round 2

Contributors: Austin

Goal:
- Place blocks inside lobby
- Implement an exit that allows players to despawn and return to the title menu

Note:
- Exit sign art and Change Color icon is still temp

What Changed:
- LobbyBlockSpawner:
    - A simple block spawner that spawns when there are fewer than 3 blocks (this value is configurable)
        - Can toggle off respawning with a checkbox if we so choose
- LobbyBlockDestroyer
    - A simple script that goes on a trigger collider which destroys blocks upon touch
        - Used to make sure blocks don't get on the ready stage and the exit area
- SquishDetector, BlockPusher, PlayerAdjacencyTracker, BlockDamager
    - Made small changes so that blocks can be pushed in the Lobby
- LobbyStateManager, PlayerSpawner
    - Added functionality so that the person who presses Play spawns into the lobby
- PlayerExitCollider
    - Despawns players and returns to title menu if the last player leaves the lobby

How to Test:
- Load up the lobby
- When you press play, you should spawn in
- Try pushing blocks around. They should behave as normal
- When there are 3 or fewer blocks in the scene, more should spawn in
- Try dying. You should respawn
- Try going to the exit. If there is more than 1 player, you should be despawned. If you are the last player, you should despawn and be taken back to the title screen
- Test in a build.
